docs/aws/audit/iamcompliance/rules/user_account_access_keys_rotation_remediation.mdx

```diff
@@ -1,6 +1,269 @@
+     - Add the Lambda function as the target for the rule.
+
+By following these steps, you can automate the process of monitoring and notifying about IAM access key rotation, thereby preventing the misconfiguration of stale access keys in your AWS environment.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the IAM console at https://console.aws.amazon.com/iam/.
+
+2. In the navigation pane, choose "Users".
+
+3. In the "User Name" list, choose the name of the desired user, which will take you to the "Summary" page for that user.
+
+4. In the "Security Credentials" section, you can see the access keys and their creation dates. If the access keys are older than 90 days, it indicates that they have not been rotated and this is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all IAM users: To check the access keys for all IAM users, you first need to list all the IAM users. You can do this by running the command `aws iam list-users`. This command will return a list of all IAM users in your AWS account.
+
+3. List access keys for each user: For each user, you need to list their access keys to check when they were last rotated. You can do this by running the command `aws iam list-access-keys --user-name <username>`. Replace `<username>` with the name of the user. This command will return a list of all access keys for the specified user, along with their creation dates.
+
+4. Check the age of each access key: To check the age of each access key, you need to compare the creation date of the key with the current date. You can do this manually, or you can write a script to automate the process. If the age of the key is more than 90 days, it is recommended to rotate the key.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. You can do this using pip:
+
+   ```
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and establish a session: In your Python script, you'll need to import the Boto3 library and establish a session with your AWS account. You'll need your AWS access key ID and secret access key.
+
+   ```python
+   import boto3
+   from datetime import datetime, timedelta
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>